src/dag/get.js

```diff
@@ -22,31 +24,20 @@ module.exports = (send) => {
     options = options || {}
     path = path || ''
 
-    if (CID.isCID(cid)) {
-      cid = cid.toBaseEncodedString()
+    try {
+      const res = ipfsPath(cid)
+      cid = res.cid
+      path = res.path || path
+    } catch (err) {
+      return callback(err)
     }
 
-    waterfall([
-      cb => {
-        send({
-          path: 'dag/resolve',
-          args: cid + '/' + path,
-          qs: options
-        }, cb)
-      },
-      (resolved, cb) => {
-        block(send).get(new CID(resolved['Cid']['/']), (err, ipfsBlock) => {
-          cb(err, ipfsBlock, resolved['RemPath'])
-        })
-      },
-      (ipfsBlock, path, cb) => {
-        if (ipfsBlock.cid.codec === 'dag-cbor') {
-          dagCBOR.resolver.resolve(ipfsBlock.data, path, cb)
-        }
-        if (ipfsBlock.cid.codec === 'dag-pb') {
-          dagPB.resolver.resolve(ipfsBlock.data, path, cb)
-        }
-      }
-    ], callback)
+    IPLDResolver.inMemory((err, ipld) => {
```

Comment:
I've been thinking that this should be done by passing the `ipfs.block.{get,put}` as the Block Service, rather than doing an inMemory resolver and then monkey patching the Block Service Instance with a fake exchange.

You should be able to do:

```
const ipld = new IPLD(ipfs.block)
```

or close

```
const ipld = new IPLD(shimBS(ipfs.block))
```